.github/workflows/e2e-hotel.yml

```diff
@@ -0,0 +1,151 @@
+name: Hotel-App End-to-End Tests
+
+on:
+  workflow_dispatch:
+#     schedule:
+#       - cron: "0 9 * * 1"
+  push:
+    branches: [main]
+    paths:
+      - "benchmarks/hotel-app/**"
+      - "utils/**"
+      - "tools/**"
+      - "runner/**"
+
+  pull_request:
+    branches: [main]
+    paths:
+      - "benchmarks/hotel-app/**"
+      - "utils/**"
+      - "tools/**"
+      - "runner/**"
+
+env:
+  GOOS: linux
+  GO111MODULE: on
+  PORT: 50051
+  KUBECONFIG: /etc/kubernetes/admin.conf
+
+jobs:
+  build-and-push:
+    name: Build and push all images
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+        with:
+          lfs: "true"
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      # - name: Install protoc
+      #   run: |
+      #     PROTOC_ZIP=protoc-3.14.0-linux-x86_64
+      #     curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.14.0/$PROTOC_ZIP.zip
+      #     unzip -o $PROTOC_ZIP.zip -d $HOME/.local
+      #     chmod +x $HOME/.local/bin/protoc
+      #     rm -f $PROTOC_ZIP.zip
+
+      - name: Build and push
+        working-directory: benchmarks/hotel-app
+        env:
+          GOPRIVATE_KEY: ${{ secrets.EASE_LAB_ACCESS_KEY }}
+        run: |
+          make all
+          make push
+
+  test-compose:
+    name: Test Docker Compose
+    needs: build-and-push
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        file:
+          [
+            dc-geo.yaml,
+            dc-geo-tracing.yaml,
+          ]
+
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+        with:
+          lfs: "true"
+
+      - name: start docker-compose benchmark
+        working-directory: benchmarks/hotel-app/yamls/docker-compose
+        run: |
+          docker-compose -f ${{ matrix.file }} pull
+          docker-compose -f ${{ matrix.file }} up &> log_file &
+          sleep 60s
+          cat log_file
+
+      - name: invoke the chain
+        run: |
+          ./tools/bin/grpcurl -plaintext -import-path ./tools/invoker/proto -proto helloworld.proto localhost:50000 helloworld.Greeter.SayHello
+
+      - name: show docker-compose log
+        run: cat benchmarks/hotel-app/yamls/docker-compose/log_file
+
+  test-knative:
+    name: Test Knative Deployment
+    needs: build-and-push
+    env:
+      GOCACHE: /root/tmp/gocache
+      GOPATH: /root/tmp/gopath
+    runs-on: [self-hosted, stock-knative]
+    strategy:
```

Comment:
Here I think we should add 
```yaml
matrix:
        service:
          [
            kn-geo,
            kn-geo-tracing,
          ]
```
Because that will help us in knative logs later on where there is a variable number of containers (because of relay, database , etc.).